compose.yml

```diff
@@ -1,20 +1,73 @@
+# Docker compose file that allows each of the services we require to start
+# only when their dependent service(s) have started. This is achieved through
+# the use of a docker healthcheck for each service.
+
+# First, construct private network for our services to communicate with each other
+networks:
+  network:
+    driver: bridge
+
 services:
-  neo4j-dbms:
-    build: "data/"
+  # neo4j service
+  neo4j-db:
+    image: neo4j:latest
+    environment:
+      NEO4J_AUTH: neo4j/6ti_qvusR!p37WwszTLvzrKbR
+    networks:
+      - network
+    # network ports, we expose non-standard ports so that we
+    # don't conflict with other neo4j instances that may be running
     ports:
-      - 7474:7474
-      - 7687:7687
-    env_file:
-      - "data/.env"
+      - 8474:7474 # HTTP for Neo4j Browser
+      - 8687:7687 # Bolt port
```

Comment:
Is there a reason why these don't map to the same port? I think ports `7474` and `7687` were the standard ports for neo4j.

Reply:
Yes, this exposes neo4j to the external system on 8xxx so if you have another neo inbstance running you wont get a port clash.

This was the reason I originally started updating the compose file as I had a conflict locally.

Reply:
I feel that running our neo4j on non-default ports could cause confusion to users familiar with neo4j.

If we want to do this, maybe this should be configurable (rather than hard-coding to non default ports)? That way, the user has the option to choose non-default ports if they really want to run a 2nd instance of neo4J on the same machine.

This feels like an edge case. I assume in most cases the user will only be running 1 neo4j per machine so the default ports will normally be fine.